[WPE][WebDriver] WebKitWebView warn if current WebContext doesn't allow automation for automated views

lauromoura opened the pull request:
#### 1a21c42e7e1c29e3e1242460fdc2cb7cd578ec8a
<pre>
[WPE][WebDriver] WebKitWebView warn if current WebContext doesn&apos;t allow automation for automated views
<a href="https://bugs.webkit.org/show_bug.cgi?id=276225">https://bugs.webkit.org/show_bug.cgi?id=276225</a>

Reviewed by Carlos Garcia Campos.

Put just a warning for now, to avoid breaking current behavior. In a
future commit we can enforce this behavior and add a capability/flag to
allow the WebDriver client request a persistent data directory (e.g. for
repeatedly testing heavy PWA or pages with huge resources)

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewConstructed): Add warning
* Tools/MiniBrowser/wpe/main.cpp:
(activate): Enable automation before creating the first view

Canonical link: <a href="https://commits.webkit.org/280695@main">https://commits.webkit.org/280695@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff828a130665be45ad9464536b14a3c5798e0426

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57330 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36658 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9805 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60952 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7773 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59458 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44282 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7963 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46424 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5493 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59360 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34398 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49508 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27287 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31180 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6819 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6778 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/53207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7090 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62631 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1243 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7184 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53685 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1248 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49542 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53764 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12676 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1055 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32487 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33572 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34657 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33318 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
```diff
@@ -905,8 +905,11 @@ static void webkitWebViewConstructed(GObject* object)
-    if (priv->isControlledByAutomation)
+    if (priv->isControlledByAutomation) {
+        if (!webkit_web_context_is_automation_allowed(priv->context.get()))
+            g_warning("WebKitWebView is-controlled-by-automation set but automation is not allowed in the context, falling back to default session.");
```
I think this should be a g_critical

lauromoura committed:
[WPE][WebDriver] WebKitWebView warn if current WebContext doesn't allow automation for automated views
https://bugs.webkit.org/show_bug.cgi?id=276225

Reviewed by Carlos Garcia Campos.

Put just a warning for now, to avoid breaking current behavior. In a
future commit we can enforce this behavior and add a capability/flag to
allow the WebDriver client request a persistent data directory (e.g. for
repeatedly testing heavy PWA or pages with huge resources)

* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewConstructed): Add warning
* Tools/MiniBrowser/wpe/main.cpp:
(activate): Enable automation before creating the first view

Canonical link: https://commits.webkit.org/280695@main
